src/lazy/value.rs

```diff
@@ -260,6 +263,52 @@ impl<'top, D: Decoder> LazyValue<'top, D> {
     pub fn read(&self) -> IonResult<ValueRef<'top, D>> {
         self.expanded_value.read_resolved()
     }
+
+    pub(crate) fn context(&self) -> EncodingContextRef<'top> {
+        self.expanded_value.context()
+    }
+
+    pub fn to_owned(&self) -> LazyElement<D> {
```

Comment:
🪧 This method is key to the whole API. By realizing it boiled down to this, I avoided needing to make a big family of `LazyElement` types to mirror the `LazyValue` family of types.